07.ls_object/lib/screen.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+require 'io/console/size'
+
+class Screen
+  def initialize(src_data, options = {})
+    @src_data = src_data
+
+    @long_format = options[:long_format]
```

Comment:
```suggestion
    @formatter = @long_format ? DetailFormatter.new(...) : NormalFormatter.new(...)
```

みたいな感じで、クラスを分けるとロジックが明確に分離されてコードの見通しがよくなるかも。
`@long_format`かどうかの判定はここ以外ではやらないイメージです。

デザインパターンのストラテジーパターンを調べてみてください。